isodatetime/parser_spec.py

```diff
@@ -171,69 +171,69 @@
 _DATE_TRANSLATE_INFO = [
     (r"\+(?=X)", "(?P<year_sign>[-+])",
      "%(year_sign)s", "year_sign"),
-    ("CC", r"(?P<century>\d\d)",
+    (r"CC", r"(?P<century>[0-9][0-9])",
      "%(century)02d", "century"),
-    ("YY", r"(?P<year_of_century>\d\d)",
+    (r"YY", r"(?P<year_of_century>[0-9][0-9])",
      "%(year_of_century)02d", "year_of_century"),
-    ("MM", r"(?P<month_of_year>\d\d)",
+    (r"MM", r"(?P<month_of_year>[0-9][0-9])",
      "%(month_of_year)02d", "month_of_year"),
-    ("DDD", r"(?P<day_of_year>\d\d\d)",
+    (r"DDD", r"(?P<day_of_year>[0-9][0-9][0-9])",
      "%(day_of_year)03d", "day_of_year"),
-    ("DD", r"(?P<day_of_month>\d\d)",
+    (r"DD", r"(?P<day_of_month>[0-9][0-9])",
      "%(day_of_month)02d", "day_of_month"),
-    ("Www", r"W(?P<week_of_year>\d\d)",
+    (r"Www", r"W(?P<week_of_year>[0-9][0-9])",
      "W%(week_of_year)02d", "week_of_year"),
-    ("D", r"(?P<day_of_week>\d)",
+    (r"D", r"(?P<day_of_week>[0-9])",
      "%(day_of_week)01d", "day_of_week"),
-    ("z", r"(?P<year_of_decade>\d)",
+    (r"z", r"(?P<year_of_decade>[0-9])",
      "%(year_of_decade)01d", "year_of_decade"),
-    ("^---", "(?P<truncated>---)",
+    (r"^---", r"(?P<truncated>---)",
      "---", None),
-    ("^--", "(?P<truncated>--)",
+    (r"^--", r"(?P<truncated>--)",
      "--", None),
-    ("^-", "(?P<truncated>-)",
+    (r"^-", r"(?P<truncated>-)",
      "-", None)
 ]
 _TIME_TRANSLATE_INFO = [
-    ("(?<=^hh)mm", r"(?P<minute_of_hour>\d\d)",
+    (r"(?<=^hh)mm", r"(?P<minute_of_hour>[0-9][0-9])",
```

Comment:
Why move away from the `\d` shorthand?

Reply:
It did not work, and gave us backslash hell. The `\d` version is also less safe because it matches Unicode numerics in other languages.

Reply:
Yup, we used the `\d` in a `regex.sub` I think, in a field that was not for regex, but for ordinary text. Problem was that we were using that method to craft the final pattern for a new regex.

Until some versions ago it was fine, but in Python 3.6 or 3.7 (can't remember which one) instead of a warning, it blew an error if I recall correctly. And the documentation of the Python function indicates that it won't be happy with characters with the `\` that are used for regex escape.

I fixed it with an ugly hack, but @matthewrmshin reviewing found a much saner solution of simply replacing the `\d` by `[0-9]`, which also increases the security for it ignores unicode numerics :clap:

Reply:
Dam it Python, I thought this was supposed to be getting easier! Taking a scan of the `re` docs they have introduced a `re.ASCII` flag which might be useful in other examples.